entrypoint.sh

```diff
@@ -1,19 +1,71 @@
 #!/bin/sh -u
+# 3scale (operations@3scale.net)
 
-cd /tmp/
+mkdir /opt/openresty/nginx/lua/
 
-# Download config files
-curl  ${THREESCALE_ENDPOINT}/admin/api/nginx.zip?provider_key=${THREESCALE_PROVIDER_KEY} -o /tmp/nginx.zip
-unzip nginx.zip && rm /tmp/nginx.zip
+reload_openresty() {
+  echo "Reloading Openresty"
+  pkill -USR1 -o nginx
+}
 
-mkdir /opt/openresty/nginx/lua/
+download_threescale_config() {
+  echo "Downloading threescale configuration, using endpoint: ${THREESCALE_ENDPOINT}"
+  cd /tmp
+  rm -rf /tmp/${THREESCALE_PROVIDER_KEY}/
```

Comment:
Tokens FTW :-)

Let's see what needed for us to release use of tokens to all...

Andrew

On Wed, Mar 16, 2016, 20:06 Michal Cichra <notifications@github.com> wrote:

> In entrypoint.sh
> <https://github.com/3scale/docker-gateway/pull/2#discussion_r56395993>:
>
> >
> > -mkdir /opt/openresty/nginx/lua/
> > +download_threescale_config() {
> > +  echo "Downloading threescale configuration, using endpoint: ${THREESCALE_ENDPOINT}"
> > +  cd /tmp
> > +  rm -rf /tmp/${THREESCALE_PROVIDER_KEY}/
>
> Imo this is task for mktemp. Also, provider key should be treated at
> least a bit securely, so it should not be part of any path.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/3scale/docker-gateway/pull/2/files/59c87bb8b8713ffa78c6588f596e2c568a08c574#r56395993>
>